### PR TITLE
support validate-traffic <traffic-path> <spec-path> command in OAV cli

### PR DESCRIPTION
**Pull request by Ligengxin96**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -42,6 +42,8 @@ Commands:
                                             and examples present in the spec.
   validate-spec <spec-path>                 Performs semantic validation of the
                                             spec.
+  validate-traffic <traffic-path>           Validate the spec file based on the
+  <spec-path>                               traffic payload.
```
Comment (leni-msft) [inline README.md:46]: better change the description to `Validate traffic payload against the spec.`

File: lib/validate.ts [hunk 2]
```diff
@@ -287,6 +289,68 @@ export async function validateExamplesInCompositeSpec(
...
+      if (!result.responseValidationResult.isSuccessful) {
+        errors.push(...result.responseValidationResult.errors)
+      }
+
+      if (errors.length > 0) {
+        for (let error of errors) {
+          console.error(JSON.stringify(error))
+        }
+      } else {
+        console.info('Validation compelete, no errors are detected.')
```
Comment (leni-msft) [inline lib/validate.ts:343]:
```suggestion
        console.info('No errors were found.')
```

Comment (leni-msft) [reply lib/validate.ts:343]: avoid using `console`, but `log` instead

File: lib/validate.ts [hunk 3]
```diff
@@ -287,6 +289,68 @@ export async function validateExamplesInCompositeSpec(
...
+      }
+
+      return errors
+    })
+  } catch (error) {
+    console.error(error)
+    return [error]
```
Comment (leni-msft) [inline lib/validate.ts:350]: throw error

**Review by leni-msft — CHANGES_REQUESTED** [review]

File: lib/commands/validate-traffic.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+export async function handler(argv: yargs.Arguments): Promise<void> {
+  await cliSuppressExceptions(async () => {
+    log.debug(argv.toString())
+    const specPath = argv.specPath
+    const trafficPath = argv.trafficPath
+    const vOptions: validate.Options = {
+      consoleLogLevel: argv.logLevel,
+      logFilepath: argv.f,
+      pretty: argv.p
```
Comment (leni-msft) [inline lib/commands/validate-traffic.ts:18]: pretty option seems not implemented yet?

Comment (Ligengxin96) [reply lib/commands/validate-traffic.ts:18]: Others command also add pretty option.

Comment (Ligengxin96) [reply lib/commands/validate-traffic.ts:18]: Without pretty option output:

image


With pretty option output: 

image

**Comment by raych1** [comment]
@Ligengxin96 , can you create PR to develop branch?

**Comment by Ligengxin96** [comment]
> @Ligengxin96 , can you create PR to develop branch?

Sure, I already changed to develop branch.

**Comment by leni-msft** [comment]
It seems there're quite a lot regression errors? https://dev.azure.com/azure-public/azsdk/_build/results?buildId=12008&view=results

**Comment by Ligengxin96** [comment]
## Command output example

### Without pretty option output:

```json
{
  message: '{"code":"INVALID_CONTENT_TYPE","message":"Invalid Content-Type (application/octet-stream).  These are supported: application/json","jsonPathsInPayload":["$"],"pathsInPayload":["$"],"schemaPath":"","severity":1,"source":{"jsonRef":"","position":{"column":-1,"line":-1}},"documentationUrl":""}',
  level: '\x1B[31merror\x1B[39m'
}
{
  message: `{"code":"OBJECT_ADDITIONAL_PROPERTIES","message":"Additional properties not allowed: etag","jsonPathsInPayload":["$['etag']"],"pathsInPayload":["$/etag"],"schemaPath":"","severity":0,"params":["etag"],"source":{"url":"/home/oav/regression/azure-rest-api-specs/specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/components_API.json","jsonRef":"#/definitions/ApplicationInsightsComponent","position":{"column":37,"line":370}},"documentationUrl":""}`,
  level: '\x1B[31merror\x1B[39m'
}

```
### With pretty option output:
```yml
error : 
code: INVALID_CONTENT_TYPE
message: >-
  Invalid Content-Type (application/octet-stream).  These are supported:
  application/json
jsonPathsInPayload:
  - $
pathsInPayload:
  - $
schemaPath: ''
severity: 1
source:
  jsonRef: ''
  position:
    column: -1
    line: -1
documentationUrl: ''

 error : 
code: OBJECT_ADDITIONAL_PROPERTIES
message: 'Additional properties not allowed: etag'
jsonPathsInPayload:
  - $['etag']
pathsInPayload:
  - $/etag
schemaPath: ''
severity: 0
params:
  - etag
source:
  url: >-
    /home/oav/regression/azure-rest-api-specs/specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/components_API.json
  jsonRef: '#/definitions/ApplicationInsightsComponent'
  position:
    column: 37
    line: 370
documentationUrl: ''
```

File: lib/validate.ts [hunk 5]
```diff
@@ -276,6 +278,72 @@ export async function validateExamplesInCompositeSpec(
   });
 }
 
+export async function validateTrafficInSpec(
```
Comment (raych1) [inline lib/validate.ts:281]: How about rename to `validateTrafficAgainstSpec`?

File: lib/validate.ts [hunk 6]
```diff
@@ -30,6 +30,8 @@ import { getErrorsFromModelValidation } from "./util/getErrorsFromModelValidatio
 import { getSuppressions } from "./validators/suppressions";
 import { log } from "./util/logging";
 import { getInputFiles } from "./generator/util";
+import { LiveValidator } from "../lib/validators/liveValidator"
```
Comment (raych1) [inline lib/validate.ts:33]: Can you reference `../lib/liveValidation/liveValidator.ts`?

File: lib/validators/liveValidator.ts [hunk 7]
```diff
@@ -0,0 +1,984 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
```
Comment (raych1) [inline lib/validators/liveValidator.ts:1]: This file can be removed.

**Review by leni-msft — APPROVED** [review]

**Comment by raych1** [comment]
Can you bump the version and add changelog?

File: ChangeLog.md [hunk 8]
```diff
@@ -1,5 +1,9 @@
 # Change Log - oav
 
+## 07/08/2021 2021 2.6.0
```
Comment (raych1) [inline ChangeLog.md:3]: Can you help remove duplicated `2021` in first few changelogs?

**Review by raych1 — APPROVED** [review]

**Pull request merged**